File: backend/src/models/Merchandiser.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Merchandiser = sequelize.define('Merchandiser', {
    merchandiser_id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    nom: {
      type: DataTypes.STRING,
      allowNull: false
    },
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
      validate: {
        isEmail: true
      }
    },
    telephone: {
      type: DataTypes.STRING,
      allowNull: false
    },
    zone_assignee: {
      type: DataTypes.STRING,
      allowNull: false
    },
    secteurs_assignes: {
      type: DataTypes.JSONB,
      allowNull: false,
      defaultValue: []
    },
    zone_geofence: {
      type: DataTypes.JSONB,
      allowNull: true
    },
    actif: {
      type: DataTypes.BOOLEAN,
      defaultValue: true
    }
  }, {
    tableName: 'merchandisers',
    indexes: [
      {
        fields: ['zone_assignee']
      },
      {
        fields: ['email'],
        unique: true
      }
    ]
  });

  return Merchandiser;
};

File: lib/domain/entities/merchandiser.dart
import 'package:equatable/equatable.dart';

class Merchandiser extends Equatable {
  final String merchandiserId;
  final String nom;
  final String email;
  final String telephone;
  final String zoneAssignee;
  final List<String> secteursAssignes;
  final Map<String, dynamic>? zoneGeofence; // GeoJSON

  const Merchandiser({
    required this.merchandiserId,
    required this.nom,
    required this.email,
    required this.telephone,
    required this.zoneAssignee,
    required this.secteursAssignes,
    this.zoneGeofence,
  });

  @override
  List<Object?> get props => [
    merchandiserId,
    nom,
    email,
    telephone,
    zoneAssignee,
    secteursAssignes,
    zoneGeofence,
  ];
}